client/src/api/plugin/queries.tsx

```diff
@@ -0,0 +1,150 @@
+import { useMutation, useQuery, useQueryClient } from '@tanstack/react-query';
+import axios from 'axios';
+import { PluginDefinition, PluginEndpoint, PluginParameters } from '../Models';
+
+export function useGetPlugins() {
+  return useQuery<PluginDefinition[]>(
+    ['plugins'],
+    async () => {
+      const response = await axios.get<PluginDefinition[]>('/api/plugins/').catch((err) => {
+        console.error('Failing getting plugins from api', err);
+        if (import.meta.env.IQENGINE_PLUGINS) {
+          const plugins = JSON.parse(import.meta.env.IQENGINE_PLUGINS);
+          return { data: plugins };
+        } else {
+          return { data: [] };
+        }
+      });
+      return response.data;
+    },
+    {
+      placeholderData: [],
+    }
+  );
+}
+
+export function useGetPlugin(plugin: PluginDefinition) {
+  return useQuery<string[]>(
+    ['plugin', plugin.name],
+    async () => {
+      const response = await axios.get<string[]>(plugin.url);
+      // make sure we onlyt return in case of a string array
+      if (!Array.isArray(response.data)) throw new Error(`Plugin ${plugin.name} does not return a string array`);
+      return response.data;
+    },
+    {
+      enabled: !!plugin,
+    }
+  );
+}
+
+export function useGetPluginParameters(pluginUrl: string) {
+  return useQuery<PluginParameters>(
+    ['plugin', 'parameters', pluginUrl],
+    async () => {
+      const response = await axios.get<PluginParameters>(pluginUrl);
+      return response.data;
+    },
+    {
+      enabled: !!pluginUrl,
+    }
+  );
+}
+
+export function useGetPluginDetailed(plugin: PluginDefinition) {
+  return useQuery<PluginEndpoint>(
+    ['plugin', plugin.name, 'detailed'],
+    async () => {
+      // make sure we have a trailing slash  to concatenate the url
+      if (plugin.url && !plugin.url.endsWith('/')) plugin.url += '/';
```

Comment:
This feels very wrong

Reply:
I know I did feel the dame them I looked at the result of this.
What do you think it is the result of this
new URL("test",  "https://plugin.com/plugins")
and what is the result os this other function
new URL("test", "https://plugin.com/plugins/")

a tip... they are not equal.